lms/product/canvas/_plugin/course_service.py

```diff
@@ -0,0 +1,31 @@
+from lms.models import ApplicationSettings, CourseGroupsExportedFromH
+from lms.product.plugin.course_service import CourseServicePlugin
+
+
+class CanvasCoursePlugin(CourseServicePlugin):
+    def __init__(self, db_session, parsed_params):
+        self._db_session = db_session
+        self._parsed_params = parsed_params
```

Comment:
Not a fan of having parsed_params here which are a bit of an unknown shape.

With parsed_params this plugin will only work on the LTILaunch view (which it might be true forever, but it's not very clear).

I'd vote for either:
- Passing the param we care about in the factory. Then the factory only it's `parsed_params` dependant.

- Passing the whole lti_params. This seems like the same situation but lti_params means something everywhere you see it. It's a known shape. In practice this has the same limitation, only will work in LTILaunches but I much more explicit. I'd say using parsed_params outside a view it's a bit confusing in general. More here where both parsed_params and LTIParams happens to be the same (or very similar).


- Passing `custom_canvas_course_id` but take it from LTIParams in the factory. Best of both worlds?

Reply:
If all these things are in `LTIParam`s then yep! This was a direct lift and shift of the previous functionality.